Remove dataprovider in iTopModulesPhpVersionIntegrationTest::testITopModulesPhpVersion

piRGoif opened the pull request:
We had problems with this dataProvider (see 2221e054), but actually we can remove it !

What do you think @odain-cbd and @bruno-ds ?

piRGoif committed:
Remove dataprovider in \Combodo\iTop\Test\UnitTest\Integration\iTopModulesPhpVersionIntegrationTest::testITopModulesPhpVersion

odain-cbd commented on test/integration/iTopModulesPhpVersionChecklistTest.php:
```diff
@@ -106,21 +90,10 @@ public function iTopModulesPhpVersionProvider() {
-				'sExpectedVersion' => $sExpectedVersion,
-				'sPhpFile' => $sPhpFile,
-			);
+			$this->CheckItopModulePhpVersion($sExpectedVersion, $sPhpFile);
```
if one datamodel file test fails immediately without checking all the others after.

it could be interesting to collect all the failures and then report the phpunit test failed with failed file list to correct.

odain-cbd replied:
removing the dataprovider will speed up the test execution. this is a good side effect

piRGoif replied:
Indeed, this is a good idea, remove the \PHPUnit\Framework\Assert::assertRegExp call and return or throw something... I'll try this !

piRGoif replied:
I've just pushed a commit that implements that, see comment below. 
I'm solving this as it is outdated now.

piRGoif committed:
:bug: Fix ITOP_VERSION undefined constant

piRGoif committed:
Checks all modules even if one is invalid
Collects error and does one assert at the end instead of one assert for each module
The later was stopping on the first error

piRGoif commented:
The test now does only one assert at the end, so that every modules are tested on each test function run (instead of failing on the first error).

All results are stored in an array, and at the end we call an assert to check that the result array is empty.

If there is a difference, the assert message is :

> Some modules have wrong versions ! They should match /^3\.0\.\d+$/

And the actual array contains for example :

```
Array (
    'C:\Dev\wamp64\www\itop-dev/datamodels/2.x/itop-backup/module.itop-backup.php' => '3.9.0'
    'C:\Dev\wamp64\www\itop-dev/datamodels/2.x/itop-portal-base/module.itop-portal-base.php' => '2.8.0'
)
```
